airbyte/_batch_handles.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+"""Batch handle class."""
```

Comment:
This class has been refactored and moved out. 

It now handles its own file creation and deletion. Deletion is generic, and creation is passed as a function ("file_opener") to the contstructor.

This class now also tracks whether the batch handle itself has been finalized or not - which makes tracking easier and more consistent across parts of the codebase which need to know this.